src/components/item/Item.js

```diff
@@ -1,85 +1,139 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useStore } from 'store/index';
 
-const Item = props => {
+const Item = ({ product }) => {
+    const { id, img, name, price, size: inventory } = product;
     const { dispatch, state } = useStore();
-    const { size } = state;
+    const [size, setSize] = useState();
+    const [quantity, setQuantity] = useState(1);
+    const [formErrors, setFormErrors] = useState({
+      currentlyInCart: false,
+      currentlyInWishlist: false,
+      size: false,
+      quantity: false,
+      outOfStock: false,
+      insufficientStock: false
+    });
 
-    const [numberOfItems, setNumberOfItems] = useState({ number: 1, id: null });
+    const handleAddToCartClick = () => {
+      const errors = {
+        quantity: quantity < 1,
```

Comment:
your previous implementation of form validation was very imperative; meaning, modifying variables directly instead of just declaring error state as a computation of other state variables. for instance

```js
// imperative
let isWrong;
const someThing = checkSomething();
if (!someThing) {
  isWrong = true;
}

// declarative
const isWrong = !checkSomething()
// no need to declare a variable without assigning it a value, then mutating it later
```